jax/_src/interpreters/pxla.py

```diff
@@ -1683,7 +1683,10 @@ class DeviceAssignmentMismatchError(Exception):
 
 
 def _get_default_device() -> xc.Device:
-  return config.default_device.value or xb.local_devices()[0]
+  if isinstance(config.default_device.value, str):
```

Comment:
What if the string is empty? You should also add `if isinstance(config.default_device.value, str) and config.default_device.value`

or maybe disallow empty strings in config.py?

Reply:
The input validity check occurs in the function "def _validate_default_device(val):"  at line 1562 of config.py